internal.nix

```diff
@@ -303,6 +306,16 @@ rec {
         cleanArgs = builtins.removeAttrs args [ "src" "packageJson" "packageLockJson" "buildInputs" "nativeBuildInputs" "nodejs" "preBuild" "postBuild" "preInstallLinks" "githubSourceHashMap" ];
         lockfile = readLockfile packageLockJson;
 
+        pnameComputed =
+          if (pname != "") then makeValidDrvName pname
+          else if (builtins.hasAttr "name" lockfile && lockfile.name != "") then lockfile.name
+          else ""; # pname is required, throw later
+
+        versionComputed =
+          if (version != "") then version
+          else if (builtins.hasAttr "version" lockfile && lockfile.version != "") then lockfile.version
+          else "0.0.0"; # version is optional
+
```

Comment:
better names than pnameComputed and versionComputed?

Reply:
I can’t really think of a better name right now but perhaps we don’t even have to come up with one and move the code above to the assignment of version and pname below? Or turn the logic into “getVersion” and “getPname” functions and call them below.

Reply:
or use args.pname and args.version : >